arches/app/utils/frontend_configuration_utils.py

```diff
@@ -0,0 +1,221 @@
+import json
+import os
+import re
+import site
+import sys
+
+from django.conf import settings
+from django.urls import get_resolver, URLPattern, URLResolver
+from django.urls.resolvers import RegexPattern, RoutePattern, LocalePrefixPattern
+
+from arches.settings_utils import list_arches_app_names, list_arches_app_paths
+
+
+def generate_frontend_configuration():
+    try:
+        _generate_frontend_configuration_directory()
+        _generate_urls_json()
+        _generate_webpack_configuration()
+        _generate_tsconfig_paths()
+    except Exception as e:
+        # Ensures error message is shown if error encountered
+        sys.stderr.write(str(e))
+        raise e
+
+
+def _generate_frontend_configuration_directory():
+    destination_dir = os.path.realpath(
+        os.path.join(_get_base_path(), "..", "frontend_configuration")
+    )
+
+    os.makedirs(destination_dir, exist_ok=True)
+
+
+def _generate_urls_json():
+    def join_paths(*parts):
+        first, *middle, last = parts
+        segments = [first.rstrip("/")]
+        segments += [part.strip("/") for part in middle]
+        segments.append(last.lstrip("/"))
+        return "/".join(segment for segment in segments if segment)
+
+    def interpolate_route(pattern):
+        if isinstance(pattern, RoutePattern):
+            return re.sub(r"<(?:[^:]+:)?([^>]+)>", r"{\1}", pattern._route)
+        elif isinstance(pattern, RegexPattern):
+            regex = pattern._regex.strip("^").rstrip("$")
+            regex = re.sub(r"\(\?P<(\w+)>[^)]+\)", r"{\1}", regex)
+            regex = re.sub(r"\(\?:[^\)]+\)", "", regex)
+            regex = re.sub(r"\[[^\]]+\]", "", regex)
+            regex = regex.replace("\\", "")
+            regex = re.sub(r"[\^\$\+\*\?\(\)]", "", regex)
+            return regex
+        return ""
+
+    def build_urls(patterns, prefix="", namespace="", collected_urls=None):
+        if collected_urls is None:
+            collected_urls = {}
+
+        for pattern in patterns:
+            if isinstance(pattern, URLPattern):
+                full_name = f"{namespace}{pattern.name}" if pattern.name else None
```

Comment:
I noticed when dumping the urls that the django admin app correctly [namespaces its urls](https://docs.djangoproject.com/en/5.1/topics/http/urls/#url-namespaces). I'll open a follow-up issue for starting to do that in arches (and lingo & controlled-lists etc) as a good practice.

This is also mentioned in Django [tutorial 3](https://docs.djangoproject.com/en/5.1/intro/tutorial03/#namespacing-url-names).

Then, we might want `generateArchesURL` to take an optional namespace arg, but that can be a follow-up.

Reply:
Yeah namespacing URLs for projects/apps may not be what we want to do. There are several cases of using route hijacking at the project level to alter the logic of core components.